runtime/message.cpp

```diff
@@ -359,14 +360,21 @@ size_t MessageBase::encode(char *to) const
 		Presence::const_iterator fpitr(_fp.get_presence().end());
 		if (!_fp.get(pp.second->_fnum, fpitr, FieldTrait::suppress))	// some fields are not encoded until unsuppressed (eg. checksum)
 		{
+			if (to > endOfSafeEncode) {
+				throw EncodedMessageExceededBuffer(FIX8_MAX_MSG_LENGTH + HEADER_CALC_OFFSET);
+			}
```

Comment:
Shouldn't this check be **after** modification of `*to`, e.g. at the end of this scope?

Reply:
I think I understand now: `endOfSaveEncode` is still in the buffer, but writing there is dangerous as it may not fit.
Please disregard my previous comment.